[BEAM-6200] Deprecate old HadoopInputFormatIO, move it into new HadoopFormatIO

aromanenko-dev opened the pull request:
Following the goal to merge read and write parts of Hadoop Input/Output Format files, we need to move `HadoopInputFormatIO.Read` into `HadoopFormatIO` and deprecate it.

The following things were done in this PR:
- `HadoopInputFormatIO.Read` implementation moved into `HadoopFormatIO`
- `HadoopInputFormatIO` is left only as a public interface, all calls are proxied to `HadoopFormatIO`
- `HadoopInputFormatIO` is deprecated
- Tests from `hadoop-input-format` package are copied to `hadoop-input-format` and adapted accordingly
- Old tests of `hadoop-input-format` are left in place to test old deprecated functionality. 

Most changes are only copying and small refactoring, no new functionality was added.

R: @dmvk R: @timrobertson100 - please, take a look


------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [![Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

timrobertson100 commented:
For info: earliest I can look is Monday but happy to review unless someone beats me to it.

Thank you @aromanenko-dev

aromanenko-dev commented on sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java:
```diff
@@ -401,412 +380,412 @@ public void testReadingData() {
-    assertThat(displayData, hasDisplayItem("key.class", serConf.get().get("key.class")));
-    assertThat(displayData, hasDisplayItem("value.class", serConf.get().get("value.class")));
-  }
+  //  @Test
```
Actually, we have to remove all these commented tests there because they tested internal functionality of `HadoopInputFormatIO` which has been moved to `HadoopFormatIO`. So, all these tests were moved into `HadoopFormatIOReadTest`. 
Now I left the only tests in `HadoopInputFormatIOTest` that do testing of public API which still can be used by user.
I forgot to delete these tests before. Thank you for pointing this out.

aromanenko-dev commented on sdks/java/io/hadoop-format/build.gradle:
```diff
@@ -53,9 +58,30 @@ dependencies {
   testCompile project(path: ":beam-sdks-java-io-common", configuration: "shadowTest")
   testCompile project(path: ":beam-sdks-java-io-jdbc", configuration: "shadow")
-  testCompile project(path: ":beam-sdks-java-io-hadoop-input-format", configuration: "shadowTest")
+//  testCompile project(path: ":beam-sdks-java-io-hadoop-input-format", configuration: "shadowTest")
```
Yes, thanks

aromanenko-dev commented:
@dmvk I addressed your comments and pushed the changes. 

Regarding your concerns - I agree that this type of tests, that require running embedded cluster, take more time than it should be for unit-tests. In the same time, the total running time (about 1 min for all tests) is still acceptable imo and they allow to test HadoopFormat.Read with different input formats. So, I think it's not an issue in this case.
